lib/ja_serializer/dsl.ex

```diff
@@ -72,30 +72,43 @@ defmodule JaSerializer.DSL do
   end
 
   @doc false
-  defmacro __before_compile__(_env) do
+  defmacro __before_compile__(env) do
     quote do
       def __relations,  do: @relations
       def __location,   do: @location
       def __attributes, do: @attributes
+
+      unquote(define_inlined_attributes_map(env))
+    end
+  end
+
+  defp define_inlined_attributes_map(env) do
+    attributes = Module.get_attribute(env.module, :attributes)
+    conn = quote do: conn
+    struct = quote do: struct
+
+    # Construct ASL for map with keys from attributes calling the attribute fn
+    body = {:%{}, [],
+      Enum.map(attributes, fn k -> {k, {k, [], [struct, conn]}} end)
+    }
```

Comment:
Here I just "manually" generating AST for what would in normal elixir look like
```
%{
  attr1: attr1(struct, conn),
  attr2: attr2(struct, conn),
  ...
})
```

For example:
```
iex(12)> quote do: %{ key_1: key_1(1, 2) }
{:%{}, [], [key_1: {:key_1, [], [1, 2]}]}
```